lib/manageiq/network_discovery/discovery.rb

```diff
@@ -30,7 +30,7 @@ def self.scan_host(ost)
 
         if ping
           # Trigger probes
-          ost.discover_types.each do |type|
+          ost.discover_types&.each do |type|
```

Comment:
If you don't pass discover_types then this method just pings the target, but doesn't do anything and doesn't raise an Exception, and so looks like it passed.  I think maybe we should `raise ArgumentError, "must pass discovery_types"` or something instead to ensure it's passed.  Unless, just pinging is the thing we want it to do?  @agrare ?

Reply:
> Unless, just pinging is the thing we want it to do?

That's not the intention, the UI won't let you submit the form without specifying at least one hypervisor discovery type.

:+1: for raising an exception

Reply:
ook, replaced by exception